File: exercise/python/day13/model/discount.py
class Discount:
    def __init__(self, product, description: str, discount_amount: float):
        self.product = product
        self.description = description
        self.discount_amount = discount_amount

    def __eq__(self, other):
        if isinstance(other, Discount):
            return (
                    self.product == other.product and
                    self.description == other.description and
                    abs(self.discount_amount - other.discount_amount) < 0.001
            )
        return False

    def __repr__(self):
        return f"Discount(description='{self.description}', discount_amount={self.discount_amount}, product={self.product})"

File: exercise/ts/day13/src/santamarket.model/Discount.ts
import { Product } from './Product';

export class Discount {
    constructor(public product: Product, public description: string, public discountAmount: number) {}

    equals(other: Discount): boolean {
        return (
            other instanceof Discount &&
            this.product.equals(other.product) &&
            this.description === other.description &&
            Math.abs(this.discountAmount - other.discountAmount) < 0.001
        );
    }
}
